au/math_test.cc

```diff
@@ -248,6 +262,43 @@ TEST(fmod, HandlesIrrationalCommonUnit) {
     EXPECT_THAT(fmod(radians(1), degrees(57)), IsNear(degrees(0.2958), degrees(0.0001)));
 }
 
+TEST(remainder, SameAsStdRemainderForNumericTypes) {
+    EXPECT_EQ(remainder(3.5, 3), std::remainder(3.5, 3));
+    EXPECT_EQ(remainder(2.5, 3), std::remainder(2.5, 3));
+}
+
+TEST(remainder, ReturnsSameTypesAsStdRemainderForSameUnitInputs) {
+    const auto expect_consistent_with_std_remainder =
+        expect_consistent_with([](auto x, auto y) { return remainder(x, y); },
+                               [](auto x, auto y) { return std::remainder(x, y); });
+
+    expect_consistent_with_std_remainder(meters(4), meters(3));
+    expect_consistent_with_std_remainder(meters(4.f), meters(3.f));
+    expect_consistent_with_std_remainder(meters(4.), meters(3.));
+    expect_consistent_with_std_remainder(meters(4.l), meters(3.l));
+    expect_consistent_with_std_remainder(meters(4), meters(3.f));
```

Comment:
Q: Why choose these mismatched types?

Reply:
The goal of this test case is to make sure that we handle combinations of _reps_ the same way that `std::remainder` handles combinations of _types_.  Adding a few mismatched-rep examples gives us better coverage for that property.